src/pages/docs/octopus-rest-api/openid-connect/index.md

```diff
@@ -0,0 +1,111 @@
+description: External systems can use OpenID Connect with service accounts to access the Octopus API without needing to provision API keys
+navOrder: 30
+hideInThisSection: true
+---
+
+Octopus supports using [OpenID Connect (OIDC)](https://openid.net/) to access the Octopus API without needing to provision API keys.
+
+Some of the benefits of using OIDC include:
+
+- API keys do not need to be provisioned and stored in external systems, reducing the risk of unauthorized access to the Octopus API from exposed keys.
+- Access tokens granted by Octopus are short-lived, reducing the risk of unauthorized access to the Octopus API.
+- API keys do not need to be rotated manually by administrators, reducing the risk of disruption when updating to newer keys in external systems.
+
+:::div{.hint}
+Using OIDC to access the Octopus API is only supported for service accounts, to access the API for a user account please use [an API key](/docs/octopus-rest-api/how-to-create-an-api-key).
+:::
+
+Any issuer that can generate signed OIDC tokens which can be validated anonymously is supported, however first-class support for GitHub Actions is provided with the [`OctopusDeploy/login`](https://github.com/OctopusDeploy/login) action.
+
+## Getting started with GitHub Actions
+
+To get started using OIDC with GitHub Actions use the below instructions. For more information see [Using OpenID Connect with Octopus and GitHub Actions](./github-actions.md).
+
+### Create an OIDC identity for a service account
+
+The first step is to create an OIDC identity for your GitHub repository to allow workflow runs to access the Octopus API.
+
+1. Go to Configuration -> Users and either create a new service account or locate an existing one.
+2. Open the OpenID Connect section.
+3. Click the New OIDC Identity button.
+4. Select GitHub Actions as the issuer type.
+5. Enter the details of your repository and how you want to filter the workflow runs that can authenticate using OIDC.
+6. Click Save.
+
+:::figure
+![OIDC Identity for GitHub Actions](/docs/octopus-rest-api/images/oidc-identity-github-actions.png "width=500")
+:::
+
+### Add the `OctopusDeploy/login` action to your workflow
+
+After the OIDC identity for GitHub Actions has been created a snippet of the `OctopusDeploy/login` step will be provided which you can use in your workflow to configure the workflow run job to use OIDC authentication.
+
+:::figure
+!['OctopusDeploy/login' snippet](/docs/octopus-rest-api/images/oidc-github-actions-details.png "width=500")
+:::
+
+1. Click the Copy to clipboard to copy the `OctopusDeploy/login` step.
+2. Paste the `OctopusDeploy/login` step into your workflow job.
+3. Add `id-token: write` to the `permissions` on the workflow job. This is required to allow the `OctopusDeploy/login` action to request an OIDC token from GitHub to use.
+   1. Depending on the other steps you have in your workflow, you may need to add additional permissions such as `contents: read`.
```

Comment:
What are those other steps that would require me to set this permission? Is there somewhere we can just link to at GitHub?

Reply:
An example of other steps that would now require additional specific permissions would be `actions/checkout` which will require `contents: read`. I remember getting tripped up when I first started playing with oidc because of the way it works, by default you don't need to supply any permissions to checkout code, but as soon as you say you want to get the id token you do need to ask for permission to read contents because using the `permissions` property resets it back to nothing. It's a bit of a footgun. 

I'd really like to get to a place with the getting started where someone can follow it and have their workflow work first time, so it felt important to note this in some way, I wonder if having it as a callout with more detail and a link like you suggested might be helpful.